From 91b2f0df3520b1d93873e0c86c1a6634c10ec8a6 Mon Sep 17 00:00:00 2001
From: Alan Winkowski <walan@marvell.com>
Date: Wed, 6 Mar 2019 17:57:26 +0200
Subject: [PATCH 025/386] net: marvell: mvpp2: fix in port_open for MUSDK

If system is in MUSDK mode, do not write to
parser when opening the port. MUSDK does this
according to init parameters and otherwise these
values get overiden by kernel.

Change-Id: Id2c7397e3358673c50e087778a2acf6506d1133b
Signed-off-by: Alan Winkowski <walan@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/5322
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 8352db1970a2..fdb361a3a8d5 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -4553,6 +4553,9 @@ static int mvpp2_open(struct net_device *dev)
 	bool valid = false;
 	int err, cpu;
 
+	if (port->flags & MVPP22_F_IF_MUSDK)
+		goto skip_musdk_parser;
+
 	err = mvpp2_prs_mac_da_accept(port, mac_bcast, true);
 	if (err) {
 		netdev_err(dev, "mvpp2_prs_mac_da_accept BC failed\n");
@@ -4574,6 +4577,7 @@ static int mvpp2_open(struct net_device *dev)
 		return err;
 	}
 
+skip_musdk_parser:
 	/* Allocate the Rx/Tx queues */
 	err = mvpp2_setup_rxqs(port);
 	if (err) {
-- 
2.17.1

